app/code/Magento/Bundle/Model/Product/Type.php

```diff
@@ -296,7 +311,12 @@ public function getChildrenIds($parentId, $required = true)
      */
     public function getParentIdsByChild($childId)
     {
-        return $this->_bundleSelection->getParentIdsByChild($childId);
+        $cacheKey = is_array($childId) ? implode('-', $childId) : $childId;
```

Comment:
In case of array is too big it can break because of lack of memory
Please test this solution for big amount of products
https://devdocs.magento.com/guides/v2.4/config-guide/cli/config-cli-subcommands-perf-data.html